docs/server/source/production-deployment-template-tendermint/architecture.rst

```diff
@@ -0,0 +1,204 @@
+  |                                    |                                                |                                    |
+  |                                    |                                                |                                    |
+  +------------------------------------+                                                +------------------------------------+
+
+
+
+.. note::
+  The arrows in the diagram represent the client-server communication. For
+  example, A-->B implies that A initiates the connection to B.
+  It does not represent the flow of data; the communication channel is always
+  fully duplex.
+
+
+NGINX: Entrypoint and Gateway
+-----------------------------
+
+We use an NGINX as HTTP proxy on port 443 (configurable) at the cloud
+entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. HTTPS Termination: The HTTPS connection does not carry through all the way
+   to BigchainDB and terminates at NGINX for now.
+
+#. Request Routing: For HTTPS connections on port 443 (or the configured BigchainDB public api port),
+   the connection is proxied to:
+
+   #. OpenResty Service if it is a POST request.
+   #. BigchainDB Service if it is a GET request.
+
+
+We use an NGINX TCP proxy on port 27017 (configurable) at the cloud
+entrypoint for:
+
+#. Rate Limiting: We configure NGINX to allow only a certain number of requests
+   (configurable) which prevents DoS attacks.
+
+#. Request Routing: For connections on port 27017 (or the configured MongoDB
+   public api port), the connection is proxied to the MongoDB Service.
+
+
+OpenResty: API Management, Authentication and Authorization
+-----------------------------------------------------------
+
+We use `OpenResty <https://openresty.org/>`_ to perform authorization checks
+with 3scale using the ``app_id`` and ``app_key`` headers in the HTTP request.
+
+OpenResty is NGINX plus a bunch of other
+`components <https://openresty.org/en/components.html>`_. We primarily depend
+on the LuaJIT compiler to execute the functions to authenticate the ``app_id``
+and ``app_key`` with the 3scale backend.
+
+
+MongoDB: Standalone
+-------------------
+
+We use MongoDB as the backend database for BigchainDB.
+In a multi-node deployment, MongoDB members communicate with each other via the
+public port exposed by the NGINX Service.
+
+We achieve security by avoiding DoS attacks at the NGINX proxy layer and by
+ensuring that MongoDB has TLS enabled for all its connections.
+
+
+Tendermint: BFT consensus engine
+--------------------------------
+
+We use Tendermint as the backend consensus engine for BFT replication of BigchainDB.
+In a multi-node deployment, Tendermint nodes/peers communicate with each other via
+the public ports exposed by the NGINX gateway.
+
+We use port **9986** (configurable) to allow tendermint nodes to access the public keys
```

Comment:
Should we update the **Notes for Firewall Setup** page in the Appendices to mention port 9986? See appendices/firewall-notes

Reply:
Yes, I will do that.

Reply:
:+1: